omlmd/listener.py

```diff
@@ -12,19 +15,26 @@ class Listener(ABC):
     """
 
     @abstractmethod
-    def update(self, source: Any, event: Event) -> None:
+    def update(self, source: t.Any, event: Event) -> None:
         """
         Receive update event.
         """
         pass
 
 
-class Event:
+class Event(ABC):
     pass
 
 
+@dataclass
 class PushEvent(Event):
-    def __init__(self, target: str, metadata: ModelMetadata):
-        # TODO: cannot just receive yet the push sha, waiting for: https://github.com/oras-project/oras-py/pull/146 in a release.
-        self.target = target
-        self.metadata = metadata
+    response: requests.Response
+    target: str
+    metadata: ModelMetadata
+
+    @property
+    def ok(self) -> bool:
+        return self.response.status_code == 200
+
+    def get_digest(self) -> str:
+        return self.response.headers["Docker-Content-Digest"] if self.ok else ""
```

Comment:
I like the idea of a helper to find the digest, but I'm not sure what this should do when the response is not ok.

Reply:
- let's return ValueError
- 2xx please

Reply:
https://github.com/oras-project/oras-py/blob/ff9c2e28e364155de68bd653ddb4da9f6979d677/oras/provider.py#L843

Oras-Py is already checking that for us before returning. As we can see on https://github.com/oras-project/oras-py/blob/ff9c2e28e364155de68bd653ddb4da9f6979d677/oras/provider.py#L642 this function performs the same checks you specified.